mono/metadata/w32file-unix.c

```diff
@@ -1369,11 +1369,13 @@ static guint32 file_seek(gpointer handle, gint32 movedistance,
 	if(highmovedistance==NULL) {
 		offset=movedistance;
 		mono_trace (G_LOG_LEVEL_DEBUG, MONO_TRACE_IO_LAYER, "%s: setting offset to %lld (low %d)", __func__,
-			  offset, movedistance);
+				  (long long) offset, movedistance);
```

Comment:
`eglib/src/glib.h` defines `gint64` and also the right modifier for it: `G_GINT64_FORMAT`.

Thus, the format string should be `"%s: setting offset to " G_GINT64_FORMAT " (low %d)"` and then the cast shouldn't be necessary. Same goes for the `(long long unsigned)` casts below.